lib/templates.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Register Gutenberg core block editor templates.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Registers Gutenberg core block editor templates.
+ */
+function gutenberg_register_templates() {
+	if (
+		get_option( 'gutenberg-experiments' ) &&
+		! array_key_exists( 'gutenberg-full-site-editing', get_option( 'gutenberg-experiments' ) )
+	) {
+		return;
+	}
+
+	register_post_type(
+		'wp_template',
+		array(
+			'labels'       => array(
+				'name' => __( 'Templates', 'gutenberg' ),
+			),
+			'show_in_rest' => true,
+		)
+	);
+
+	$template_query = new WP_Query(
+		array(
+			'post_type' => 'wp_template',
+			'name'      => 'single-post',
+		)
+	);
+	if ( ! $template_query->have_posts() ) {
+		wp_insert_post(
+			array(
+				'post_type'    => 'wp_template',
+				'post_name'    => 'single-post',
+				'post_content' => '<!-- wp:post-title /--><!-- wp:post-content /-->',
```

Comment:
Should the featured image be factored in here?

Reply:
If that's something we want in the default "post" post type template, sure.

This shouldn't be in Core or the plugin though. We should provide the framework and UI for users to create new templates, but default templates, if there are any, should be provided by themes.